tests/class_test.py

```diff
@@ -0,0 +1,109 @@
+        self.office = Office
+
+    '''checks the class relationships'''
+
+    def test_office_is_subclass(self):
+        self.assertIsInstance(Office, Room)
+
+    def test_living_space_is_subclass(self):
+        self.assertIsInstance(LivingSpace, Room)
+
+    def test_room_is_subclass(self):
+        self.assertIsInstance(Room, Amity)
+
+    def test_staff_is_subclass(self):
+        self.assertIsInstance(Staff, Person)
+
+    def test_fellow_is_subclass(self):
+        self.assertIsInstance(Fellow, Person)
+
+    def test_create_living_space(self):
+        '''test if it creates living space and assert has 4 spaces'''
+        self.assertEqual(self.living_space.create_room, 4)
+
+    def test_create_office(self):
+        '''test if it creates office  and has 6 spaces'''
+        self.assertEqual(self.office.create_room, ["room name"])
+
+    def test_room_is_office(self):
+        '''test if room created is an office'''
+        self.assertEqual(self.test_amity.create_room, {"Valarys":"Office"})
+
+    def test_room_is_living_space(self):
+        '''test if room created is a living space'''
+        self.assertEqual(self.test_amity.create_room, {"King's Landing":"Living Space"})
+
+    def test_room_exists(self):
+        '''test if a room exists from a list of rooms'''
+        self.assertIn(self.test_amity.view_rooms, ["King's Landing"])
+
+    def test_check_office_has_space(self):
+        '''test if office has space. If list is equal to 6 the office space is full.'''
+        self.assertEqual(self.office.check_office_has_space, 6)
+
+    def test_check_living_space_has_space(self):
+        '''test if the living space has space. If the allocated is 4 it raises an assert'''
+        self.assertEqual(self.living_space.check_living_space_has_space, 4)
+
+    def test_add_person(self):
+        '''tests that person has been created'''
+        self.add_person = Person()
+        self.new_person = self.staff or self.fellow
+        self.assertEqual(self.new_person.add_person, {"emp_id": "name"} )
+
+    def test_add_staff(self):
+        '''Test adding of new staff'''
+        self.new_staff = self.staff
+        self.assertEqual(self.new_staff.add_staff, {"emp_id": "name"})
+
+    def test_add_fellow(self):
+        '''Test adding of new fellow'''
+        self.new_fellow = Fellow
+        self.assertEqual(self.new_fellow.add_fellow, {"emp_id": "name"})
+
+    def test_is_fellow(self):
+        '''Test that person is a fellow'''
+        self.a_person = Fellow()
+        self.assertTrue(self.a_person.is_fellow, msg= "Person is a fellow" )
+
+    def test_is_staff(self):
+        '''Check name doesnt exist'''
+        self.a_person = Staff
+        self.assertTrue(self.a_person.is_staff, )
+
+    def test_name_exists(self):
+        ''' test if name for room create exists '''
+        self.assertEqual(self.test_amity.view_rooms, "Name exists")
+
+    def test_check_duplicate_person_to_office(self):
```

Comment:
For these tests where you are checking for duplication, you could probably use some actual data or input to simulate actual duplication. 
Otherwise the `view_allocated` won't have any return value or logic to indicate that the Person already exists.
Same advice goes for most tests above.